rust/operator/src/kafka_controller.rs

```diff
@@ -741,3 +865,76 @@ fn build_broker_rolegroup_statefulset(
 pub fn error_policy(_error: &Error, _ctx: Arc<Ctx>) -> Action {
     Action::requeue(Duration::from_secs(5))
 }
+
+/// We only expose client HTTP / HTTPS and Metrics ports.
+fn service_ports(kafka: &KafkaCluster) -> Vec<ServicePort> {
+    let mut ports = vec![ServicePort {
+        name: Some(METRICS_PORT_NAME.to_string()),
+        port: METRICS_PORT.into(),
+        protocol: Some("TCP".to_string()),
+        ..ServicePort::default()
+    }];
+
+    if kafka.client_tls_secret_class().is_some() || kafka.client_authentication_class().is_some() {
+        ports.push(ServicePort {
+            name: Some(SECURE_CLIENT_PORT_NAME.to_string()),
+            port: SECURE_CLIENT_PORT.into(),
+            protocol: Some("TCP".to_string()),
+            ..ServicePort::default()
+        });
+    } else {
+        ports.push(ServicePort {
+            name: Some(CLIENT_PORT_NAME.to_string()),
+            port: CLIENT_PORT.into(),
+            protocol: Some("TCP".to_string()),
+            ..ServicePort::default()
+        });
+    }
+
+    ports
+}
+
+/// We only expose client HTTP / HTTPS and Metrics ports.
+fn container_ports(kafka: &KafkaCluster) -> Vec<ContainerPort> {
+    let mut ports = vec![ContainerPort {
+        name: Some(METRICS_PORT_NAME.to_string()),
+        container_port: METRICS_PORT.into(),
+        protocol: Some("TCP".to_string()),
+        ..ContainerPort::default()
+    }];
+
+    if kafka.client_tls_secret_class().is_some() || kafka.client_authentication_class().is_some() {
+        ports.push(ContainerPort {
+            name: Some(SECURE_CLIENT_PORT_NAME.to_string()),
+            container_port: SECURE_CLIENT_PORT.into(),
+            protocol: Some("TCP".to_string()),
+            ..ContainerPort::default()
+        });
+    } else {
+        ports.push(ContainerPort {
+            name: Some(CLIENT_PORT_NAME.to_string()),
+            container_port: CLIENT_PORT.into(),
+            protocol: Some("TCP".to_string()),
+            ..ContainerPort::default()
+        });
+    }
+
+    ports
+}
+
+fn create_tls_volume(volume_name: &str, tls_secret_class: Option<&TlsSecretClass>) -> Volume {
+    let mut secret_class_name = "tls";
```

Comment:
We could use something like `let secret_class_name = tls_secret_class.and_then(|t| t.secret_class).unwrap_or("tls")` (not tested)

Reply:
done